packages/e2e-tests/specs/editor/blocks/site-title.test.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	createNewPost,
+	insertBlock,
+	pressKeyWithModifier,
+	switchUserToAdmin,
+	switchUserToTest,
+	visitAdminPage,
+} from '@wordpress/e2e-test-utils';
+
+async function getSetting( setting ) {
+	await switchUserToAdmin();
+	await visitAdminPage( 'options-general.php' );
+
+	const value = await page.$eval(
+		`#${ setting }`,
+		( element ) => element.value
+	);
+	await switchUserToTest();
+	return value;
+}
+
+async function setSetting( setting, value ) {
+	await switchUserToAdmin();
+	await visitAdminPage( 'options-general.php' );
+
+	await page.focus( `#${ setting }` );
+	await pressKeyWithModifier( 'primary', 'a' );
+	await page.type( `#${ setting }`, value );
+
+	await Promise.all( [
+		page.click( '#submit' ),
+		page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+	] );
+
+	await switchUserToTest();
+}
+
+const saveEntities = async () => {
+	const savePostSelector = '.editor-post-publish-button__button';
+	const savePanelSelector = '.entities-saved-states__panel';
+	const entitiesSaveSelector = '.editor-entities-saved-states__save-button';
+	const publishPanelSelector = '.editor-post-publish-panel';
+	const closePanelButtonSelector =
+		'.editor-post-publish-panel__header-cancel-button button';
+
+	await page.click( savePostSelector );
+	await page.waitForSelector( savePanelSelector );
+	await page.click( entitiesSaveSelector );
+	await page.waitForSelector( publishPanelSelector );
+	await page.click( closePanelButtonSelector );
+};
+
+describe( 'Site Title block', () => {
+	let originalSiteTitle;
+	beforeAll( async () => {
+		originalSiteTitle = await getSetting( 'blogname' );
+	} );
+
+	afterAll( async () => {
+		await setSetting( 'blogname', originalSiteTitle );
+	} );
+
+	it( 'Can edit the site title', async () => {
+		await createNewPost();
+		await insertBlock( 'Site Title' );
+		const editableSiteTitleSelector =
+			'.wp-block-site-title a[contenteditable="true"]';
+		await page.waitForSelector( editableSiteTitleSelector );
+		await page.focus( editableSiteTitleSelector );
+		await pressKeyWithModifier( 'primary', 'a' );
+
+		// Create a second list item.
+		await page.keyboard.type( 'New Site Title' );
+
+		await saveEntities();
+
+		await page.reload();
```

Comment:
Since we're getting the `title` from `admin settings page` this seems redundant, but if we remove this it fails in `non-interactive` mode. That feels it could lead to intermittent failures, no?

Reply:
Interesting 🤔  I have to admit that I'd only run this in interactive mode so far, so I didn't experience the failures.

I scattered a few `page.screenshot()` calls to aid debugging, and I think I've tracked down the issue:

![save](https://user-images.githubusercontent.com/96308/122920620-e3149f00-d361-11eb-951b-1b154b2a0d9e.jpg)

This is right after clicking 'Save' (and before closing the pre-publish panel via 'Cancel'). The 'Publish' and 'Cancel' buttons might be already there, but it looks like the network request is still being processed.

The fix is likely to add a `waitForSelector` for the some pre-publish panel content; or (outside of e2e tests) even to disable the 'Publish' and 'Cancel' button while we're still processing.

Reply:
Bingo:

```diff
diff --git a/packages/e2e-tests/specs/editor/blocks/site-title.test.js b/packages/e2e-tests/specs/editor/blocks/site-title.test.js
index 8c861dbbc9..a016199ef3 100644
--- a/packages/e2e-tests/specs/editor/blocks/site-title.test.js
+++ b/packages/e2e-tests/specs/editor/blocks/site-title.test.js
@@ -50,6 +50,9 @@ const saveEntities = async () => {
 	await page.waitForSelector( savePanelSelector );
 	await page.click( entitiesSaveSelector );
 	await page.waitForSelector( publishPanelSelector );
+	await page.waitForSelector(
+		'.editor-post-publish-panel__header-cancel-button button:not([disabled])'
+	);
 	await page.click( closePanelButtonSelector );
 };
 
@@ -76,7 +79,6 @@ describe( 'Site Title block', () => {
 
 		await saveEntities();
 
-		await page.reload();
 		const siteTitle = await getSetting( 'blogname' );
 		expect( siteTitle ).toEqual( 'New Site Title' );
 	} );
diff --git a/packages/editor/src/components/post-publish-panel/index.js b/packages/editor/src/components/post-publish-panel/index.js
index c2092bd70e..eae0e36add 100644
--- a/packages/editor/src/components/post-publish-panel/index.js
+++ b/packages/editor/src/components/post-publish-panel/index.js
@@ -94,10 +94,15 @@ export class PostPublishPanel extends Component {
 									onSubmit={ this.onSubmit }
 									forceIsDirty={ forceIsDirty }
 									forceIsSaving={ forceIsSaving }
+									isSaving={ isSaving }
 								/>
 							</div>
 							<div className="editor-post-publish-panel__header-cancel-button">
-								<Button onClick={ onClose } variant="secondary">
+								<Button
+									disabled={ isSaving }
+									onClick={ onClose }
+									variant="secondary"
+								>
 									{ __( 'Cancel' ) }
 								</Button>
 							</div>

```

I'll file a separate PR for the button changes, since they deserve more visibility.

Reply:
BTW here's the diff that I used to create the screenshot:

```diff
diff --git a/packages/e2e-tests/specs/editor/blocks/site-title.test.js b/packages/e2e-tests/specs/editor/blocks/site-title.test.js
index a016199ef3..49804f1134 100644
--- a/packages/e2e-tests/specs/editor/blocks/site-title.test.js
+++ b/packages/e2e-tests/specs/editor/blocks/site-title.test.js
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+import path from 'path';
+
 /**
  * WordPress dependencies
  */
@@ -50,6 +55,9 @@ const saveEntities = async () => {
        await page.waitForSelector( savePanelSelector );
        await page.click( entitiesSaveSelector );
        await page.waitForSelector( publishPanelSelector );
+       await page.screenshot( {
+               path: path.join( process.cwd(), 'artifacts', 'saveEntities.jpg' ),
+       } );
        await page.waitForSelector(
                '.editor-post-publish-panel__header-cancel-button button:not([disabled])'
        );
```

Reply:
> I'll file a separate PR for the button changes, since they deserve more visibility.

https://github.com/WordPress/gutenberg/pull/32889